Feature

vipozdn opened the pull request:
## Checklist

__Common__

- [x] I am ran the app before creating PR
- [x] I am ran all tests before creating PR

__UI__

- [x] I am ran the app for visual checks.

__Logic__

- [x] I am tested basic app functionality.

## Changes

Describe all changes here:

- First;
- Second;
- Third;
- ...

## Comments

Describe all additional information here.

vipozdn committed:
2021-05-12 v. 1.0.8: added openGooglePlay method

1. Added:
- method that opens Google Play link in WebUtils class

vipozdn committed:
2021-05-12 v. 1.0.8: added RateApp support methods

1. Added:
- canShowRateAppDialog, increaseHashGenerationCount, save and
getIntPreference methods to SharedPreferencesSettingsHelper class;
- key_hash_generation_count string to strings-non-translatable.

vipozdn committed:
2021-05-12 v. 1.0.8: added rate AppAlertDialog

1. Added:
- AppAlertDialog that shows Rate App dialog in HashCalculatorFragment;
- rate app strings to strings.xml.

fartem commented on app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/logic/hashcalculator/ui/HashCalculatorFragment.java:
```diff
@@ -97,6 +100,24 @@
             );
         } else {
             etGeneratedHash.setText(hashValue);
+            SharedPreferencesSettingsHelper settingsHelper = new SharedPreferencesSettingsHelper(context);
```
You need to use `SettingsHelper` instance from Dagger injection. If a class don't contains `SettingsHelper` injection, add it as local public field with `@Inject` annotation.

fartem reviewed (changes requested):
Update `SettingsHelper` injection in `HashCalculatorFragment`.

fartem commented:
@vipozdn Also, you can copy instrumentation tests for Rate dialog from [here](https://github.com/hash-checker/hash-checker/blob/master/app/src/androidTest/java/com/smlnskgmail/jaman/hashchecker/rateapp/RateAppDialogTest.java).

vipozdn committed:
2021-05-13 v. 1.0.8: added Dagger injection usage

1. Added:
- usage of SettingsHelper instance from Dagger injection in HashCalculatorFragment;
- declaration of methods from SharedPreferencesSettingsHelper class to SettingsHelper interface;
- RateAppDialogTest class extending BaseUITest.

fartem commented on app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/logic/settings/api/SettingsHelper.java:
```diff
@@ -41,4 +43,7 @@
 
     void setRefreshSelectedFileStatus(boolean status);
 
+    boolean canShowRateAppDialog(Context context);
```
Delete `Context` argument from `canShowRateAppDialog` and `increaseHashGenerationCount` methods. Context available in interface realization (`SharedPreferencesSettingsHelper`).

vipozdn replied:
okay!

fartem commented:
@vipozdn Also, check [Codacy report](https://app.codacy.com/gh/hash-checker/hash-checker-lite/pullRequest?prid=7422198). In last commit you have some code issues.

vipozdn committed:
2021-05-16 v. 1.0.8: fixed codacy report issue

3. Updated:
- documentation by fartem.

4. Deleted:
- unused import at HashCalculatorFragment class.

vipozdn committed:
2021-05-16 v. 1.0.8: deleted Context argument

1. Deleted:
- Context argument from canShowRateAppDialog and increaseHashGenerationCount methods at SettingsHelper, SharedPreferencesSettingsHelper and HashCalculatorFragment;
- unused import at SettingsHelper class.

fartem commented:
@vipozdn thanks.

vipozdn commented:
> 
> 
> @vipozdn thanks.

I'm always happy to help!

fartem commented:
@vipozdn Also, your PR contains some issues from [CPD](https://pmd.github.io/latest/pmd_userdocs_cpd.html). You can fix issues and assign [this](https://github.com/hash-checker/hash-checker-lite/issues/70) issue. CPD checks run with `cpdCheck` task by Gradle.

vipozdn commented:
> @vipozdn Also, your PR contains some issues from [CPD](https://pmd.github.io/latest/pmd_userdocs_cpd.html). You can fix issues and assign [this](https://github.com/hash-checker/hash-checker-lite/issues/70) issue. CPD checks run with `cpdCheck` task by Gradle.

Okay
